### PR TITLE
add ident-format option, fix identificator bugs

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -21,6 +24,41 @@ pub const BITS_PER_BYTE: u32 = 8;
 /// that are not valid in Rust ident
 const BLACKLIST_CHARS: &[char] = &['(', ')', '[', ']', '/', ' ', '-'];
 
+fn to_pascal_case(s: &str) -> String {
+    if !s.contains('_') {
+        s.to_pascal_case()
+    } else {
+        let mut string = String::new();
+        let mut parts = s.split('_').peekable();
```
Comment (Emilgardis) [inline src/util.rs:32]: Should this maybe first convert to UPPER_CASE, and then convert to this "special" pascal case?

**Review by Emilgardis — CHANGES_REQUESTED** [review]

File: src/util.rs (hunk 1 above)
Comment (Emilgardis) [reply src/util.rs:32]: Sorry, missed this in #805

Comment (burrbull) [reply src/util.rs:32]: For `_1token` case? Maybe.

Comment (burrbull) [reply src/util.rs:33]: No. Same result.

Comment (burrbull) [reply src/util.rs:32]: I don't see other issues except this. And this name does not look correct for ident.
It's better to ping someone from ESP to fix SVD.

Comment (Emilgardis) [reply src/util.rs:32]: ah, I was mistaken about what the code does, this looks fine

**Review by Emilgardis — APPROVED** [review]

**Pull request merged**